contracts/utils/math/Math.sol

```diff
@@ -229,6 +249,26 @@ library Math {
         return mulDiv(x, y, denominator) + SafeCast.toUint(unsignedRoundsUp(rounding) && mulmod(x, y, denominator) > 0);
     }
 
+    /**
+     * @dev Calculates floor(x * y >> n) with full precision. Throws if result overflows a uint256.
+     */
+    function mulShr(uint256 x, uint256 y, uint8 n) internal pure returns (uint256 result) {
+        unchecked {
+            (uint256 prod1, uint256 prod0) = mul512(x, y);
+            if (prod1 >= 1 << n) {
+                Panic.panic(Panic.UNDER_OVERFLOW);
+            }
+            return (prod1 << (256 - n)) | (prod0 >> n);
```

Comment:
I like `high`, `low` better as used in the function definition.

Reply:
For reference, prod1 and prod0 has been historically used in mulDiv 

Would you also change that part ?

Reply:
Let's be consistent with whatever we choose. I think high and low is better for readability, do you agree? If so let's change that too.

Reply:
changed